### PR TITLE
interactive: add "copy query" button (6/7)

### PR DESCRIPTION
**Pull request by dshaaban01**
add "copy query" button to allow the user to copy the query that he has generated. 

image

### CODE REVIEW

File: xdsl/interactive/app.tcss [hunk 1]
```diff
@@ -21,12 +28,12 @@
...
 
-# Horizontal( SelectionList, ScrollableContainer(Label) )
+# Horizontal( SelectionList, Horizontal(Button), ScrollableContainer(Label) )
 #selected_passes_and_list_horizontal{
     height: 100%;
     layout: grid;
-    grid-size: 2 1;
-    grid-columns: 50% 50%;
+    grid-size: 3 1;
+    grid-columns: 50% 10% 40%;
```
Comment (superlopuh) [inline xdsl/interactive/app.tcss:36]: Would it make sense to keep the grid be 50-50, and then the buttons that act on either the pass pipeline builder or pass pipeline representaiton be within their section of the grid? the 10% seems a bit arbitrary, since the buttons are of a fixed dimension.

Comment (superlopuh) [reply xdsl/interactive/app.tcss:36]: Could you please give this a go? There should be a way to say that the button has whatever size it wants.

**Review by superlopuh — CHANGES_REQUESTED** [review]

**Review by webmiche — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -101,6 +101,16 @@ exclude = """
 |xdsl/_version.py$
 """
 
+[tool.flake8]
```
Comment (superlopuh) [inline pyproject.toml:104]: please remove this

File: tests/test_mlctx.py [hunk 3]
```diff
@@ -92,21 +92,3 @@ def test_get_attr_unregistered(is_type: bool):
     )
     if is_type:
         assert issubclass(attr, TypeAttribute)
-
-
-def test_clone_function():
```
Comment (superlopuh) [inline tests/test_mlctx.py:97]: undo this change, please

File: xdsl/ir/core.py [hunk 4]
```diff
@@ -69,17 +69,9 @@ class MLContext:
...
-    _loaded_attrs: dict[str, type[Attribute]] = field(default_factory=dict)
-
-    def clone(self) -> MLContext:
-        return MLContext(
-            self.allow_unregistered,
-            self._loaded_dialects.copy(),
-            self._loaded_ops.copy(),
-            self._loaded_attrs.copy(),
-        )
+    _loaded_dialects: dict[str, Dialect] = field(init=False, default_factory=dict)
```
Comment (superlopuh) [inline xdsl/ir/core.py:72]: ditto

**Review by compor — APPROVED** [review]
LGTM minus the changes that have crept in from previous PRs.

File: tests/test_mlctx.py
Comment (superlopuh) [inline tests/test_mlctx.py:1]: can you rename this back, please?

File: tests/test_mlcontext.py
Comment (superlopuh) [inline tests/test_mlcontext.py:1]: now this file is deleted :)

File: xdsl/interactive/app.py [hunk 5]
```diff
@@ -187,6 +186,15 @@ def copy_output(self, event: Button.Pressed) -> None:
         """When the "Copy Output" button is pressed, the output IR TextArea is copied"""
         pyclip_copy(self.output_text_area.text)
 
+    @on(Button.Pressed, "#copy_query_button")
+    def copy_query(self, event: Button.Pressed) -> None:
+        """When the "Copy Query" button is preseed, the selected passes/query is copied"""
```
Comment (superlopuh) [inline xdsl/interactive/app.py:191]:
```suggestion
        """When the "Copy Query" button is preseed, the selected passes/query is copied."""
```

**Review by superlopuh — APPROVED** [review]
